### PR TITLE
Parse rban to new classes

### PR DESCRIPTION
**Pull request by ilianolhin**
I just made it work, without refactoring existing code. I have neither time, nor energy for refactoring now.

### CODE REVIEW

File: src/nerpa_pipeline/handle_rban.py [hunk 1]
```diff
@@ -279,60 +313,41 @@ def rban_postprocessing(path_to_rban_output, main_out_dir, path_to_rban, path_to
...
-        try to recognize some unidentified monomers by removing a lipid tail from them and then running rban again on the trimmed monomers
-        the initial rban results are unaffected (and used further), only the dictionary of newly recognized monomers is created
-        '''
+        # try to recognize some unidentified monomers by removing a lipid tail from them and then running rban again on the trimmed monomers
+        # the initial rban results are unaffected (and used further), only the dictionary of newly recognized monomers is created
         hybrid_monomers_dict = rban_postprocessing(path_to_rban_output, main_out_dir, path_to_rban, path_to_monomers_db, log)
 
+    nrp_variants = []
+    graph_records = []
     with open(path_to_graphs, 'w') as f_out:
```
Comment (azmtag) [inline src/nerpa_pipeline/handle_rban.py]: looks obsolete

Comment (IlyaOlkhovsky) [reply src/nerpa_pipeline/handle_rban.py:338]: Why?

Comment (ilianolhin) [reply src/nerpa_pipeline/handle_rban.py]: Thanks, fixed

File: src/nerpa_pipeline/handle_rban.py [hunk 2]
```diff
@@ -105,41 +111,102 @@ def parse_as_simple_cycle(G: nx.DiGraph) -> Union[List[int], None]:
...
-            cycles.append(cycle)
+            backbone_sequences.append(BackboneSequence(type='CYCLE',
+                                                       node_idxs=cycle))
         else:
-            paths += [ham_path for u in Gs
-                      if (ham_path := hamiltonian_path(Gs, u))]
+            backbone_sequences.extend(BackboneSequence(type='PATH',
+                                                       node_idxs=ham_path)
+                                      for u in Gs
+                                      if (ham_path := hamiltonian_path(Gs, u)))
```
Comment (azmtag) [inline src/nerpa_pipeline/handle_rban.py]: Potentially can lead to several paths per component (eg. when rban fails to detach two monomers)

File: src/nerpa_pipeline/handle_rban.py [hunk 3]
```diff
@@ -154,76 +221,43 @@ def add_hybrid_monomers(graph):
...
-    return structures
+    # all fragments together
+    if len(backbones) > 1:
+        variants.append(NRP_Variant(nrp_id=rban_record['id'],
+                                    fragments=list(map(backbone_to_fragment, backbones))))
+
+    return graph_record, variants
 
 
 def rban_postprocessing(path_to_rban_output, main_out_dir, path_to_rban, path_to_monomers_db, log):
```
Comment (azmtag) [inline src/nerpa_pipeline/handle_rban.py:279]: I didn't notice in the prev pr: "This function tries to remove a lipid tail" should read as "This function tries to remove a polyketide chain"

File: src/nerpa_pipeline/handle_rban.py [hunk 4]
```diff
@@ -279,60 +313,40 @@ def rban_postprocessing(path_to_rban_output, main_out_dir, path_to_rban, path_to
 
 
 def generate_info_from_rban_output(path_to_rban_output, path_to_monomers_tsv, path_to_graphs,
```
Comment (azmtag) [inline src/nerpa_pipeline/handle_rban.py]: I think we need to rename this function :)
And change the help string

File: src/nerpa_pipeline/handle_rban.py [hunk 5]
```diff
@@ -279,60 +313,40 @@ def rban_postprocessing(path_to_rban_output, main_out_dir, path_to_rban, path_to
...
     log.info(f'results will be in {path_to_graphs}', indent=1)
     recognized_monomers = [x.split()[0] for x in open(path_to_monomers_tsv)]
     recognized_monomers = set(recognized_monomers[1:])  # reading the set of monomers supported by nerpa from the config file ([1:] because first row is annotations)
     hybrid_monomers_dict = defaultdict(dict)
     if process_hybrids:
-        '''
-        try to recognize some unidentified monomers by removing a lipid tail from them and then running rban again on the trimmed monomers
-        the initial rban results are unaffected (and used further), only the dictionary of newly recognized monomers is created
-        '''
+        # try to recognize some unidentified monomers by removing a lipid tail from them and then running rban again on the trimmed monomers
```
Comment (azmtag) [inline src/nerpa_pipeline/handle_rban.py]: "by identifying and removing a polyketide chain"

**Pull request merged**